From 1283dcd736f2ab0b0b9463d78794ec615ad8584a Mon Sep 17 00:00:00 2001
From: Amit Kumar Mahapatra <amit.kumar-mahapatra@xilinx.com>
Date: Mon, 31 Aug 2020 02:27:42 -0600
Subject: [PATCH 1542/1852] spi: spi-zynq-qspi: Simplify logic for slave select

commit 6b0f308b079fbd9ef7a840950e73128285c892fc from
https://github.com/Xilinx/linux-xlnx.git

Simplified the logic to assert or de-assert the chip select line by
either setting or resetting the concerned bit.

Addresses-Coverity: "integer_truncated"
Signed-off-by: Amit Kumar Mahapatra <amit.kumar-mahapatra@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/spi/spi-zynq-qspi.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/drivers/spi/spi-zynq-qspi.c b/drivers/spi/spi-zynq-qspi.c
index 51e4c5c7e2cd..cc31f4b29f09 100644
--- a/drivers/spi/spi-zynq-qspi.c
+++ b/drivers/spi/spi-zynq-qspi.c
@@ -381,11 +381,7 @@ static void zynq_qspi_chipselect(struct spi_device *spi, bool is_high)
 	} else {
 		/* Select the slave */
 		config_reg &= ~ZYNQ_QSPI_CONFIG_SSCTRL_MASK;
-		if (gpio_is_valid(spi->cs_gpio)) {
-			config_reg |= (((~(BIT(0))) <<
-					ZYNQ_QSPI_SS_SHIFT) &
-					ZYNQ_QSPI_CONFIG_SSCTRL_MASK);
-		} else {
+		if (!gpio_is_valid(spi->cs_gpio)) {
 			config_reg |= (((~(BIT(spi->chip_select))) <<
 					ZYNQ_QSPI_SS_SHIFT) &
 					ZYNQ_QSPI_CONFIG_SSCTRL_MASK);
-- 
2.31.1

